x-pack/test/fleet_api_integration/apis/epm/bulk_upgrade.ts

```diff
@@ -51,13 +51,15 @@ export default function (providerContext: FtrProviderContext) {
           .post(`/api/fleet/epm/packages/_bulk`)
           .auth(testUsers.fleet_all_int_read.username, testUsers.fleet_all_int_read.password)
           .set('kbn-xsrf', 'xxxx')
+          .send({ packages: ['multiple_versions', 'overrides'] })
```

Comment:
The reason for these having to be updated to include a valid `body` is because Authz is now being done in the handler (or handler wrapper) which is after the core HTTP framework validates the request using the route's schema. Without these changes, we were getting `400` errors

Reply:
Should we not do authorization before validation?

Reply:
Hi @nchaulet ,

Ideally, yes it should. However, the Kibana Router framework does not provide for an easy way to do that. They have middleware that we can register for post Auth/pre validation, but it does not provide you with enough information to be able do authz - specifically: we would need access to the route registration configuration, especially the information around what `path` pattern the route was registered with. 

I actually took a note to create a post to Kibana security slack channel asking them to support a `authz(): boolean | Promise<Boolean>` option on route registration. Something like:

```typescript
 router.post(
    {
      path: '/api/someplugin/{token}/view',

      authz: (): boolean => { }

    },
    async () => {}
  );
```

If have them call `authz` automatically after `auth` is done if `auths` was defined on the